cve_bin_tool/parsers/python.py

```diff
@@ -28,16 +28,22 @@ def run_checker(self, filename):
                     "--report",
                     "-",
                     "--quiet",
-                ]
+                ],
+                stderr=subprocess.STDOUT,
             )
-        )
-        for line in lines["install"]:
-            product = line["metadata"]["name"]
-            version = line["metadata"]["version"]
-            vendor = self.find_vendor(product, version)
-            if vendor is not None:
-                yield from vendor
-        self.logger.debug(f"Done scanning file: {self.filename}")
+        except subprocess.CalledProcessError as e:
+            self.logger.error(f"Status : FAIL {e.returncode} {e.output}")
```

Comment:
Could you please elaborate on what you mean by report and how can I access it?

Reply:
If you look up a few lines in the code, you can see how pip is being run:

```
                    "pip3",
                    "install",
                    "-r",
                    self.filename,
                    "--dry-run",
                    "--ignore-installed",
                    "--report",
                    "-",
                    "--quiet",
```

When I run that same command manually, you'll get the "report" in question.  

When I run it, it generates a warning about report not yet producing a stable API.  It's possible my version of pip is a few weeks old and this has been fixed.  But basically if they change pip's output, we may need to change our code, so if you happened to get a version that was a bit different it could explain some test breakages.  Or it could be unrelated.  I just wanted to make sure you knew that we can't currently assume that what we get from pip won't change.

Reply:
And if it *is* fixed and pip says they've got some sort of stable report since I last checked in December, we probably want to explicitly make that version-or-later of pip a requirement.

Reply:
Yes, for me as well, report is giving that warning `--report is currently an experimental option. The output format may change in a future release without prior warning` .

Reply:
Perhaps we can make pip upgrade to the latest version if the error is thrown?

Reply:
And later if report is removed in a future version of pip, we could make pip to the last pip version which supported report.

Reply:
> Perhaps we can make pip upgrade to the latest version if the error is thrown?

I don't think it's a good idea to update anything during execution, it's an unusual and undesirable side effect.

> And later if report is removed in a future version of pip, we could make pip to the last pip version which supported report.

I doubt it will be removed anytime soon, it's not even stable yet.

`pip` is a very important part of the Python ecosystem, doing any manipulations with it at runtime and even just limiting its version as a `cve-bin-tool` dependency is highly undesirable IMO.

Reply:
I agree with @Molkree - we MUST not auto-update any of the installed software. Given the fact that the `--report` option is still experimental, I wonder if we should be (a) checking that the installed version of pip supports the option (although this could be simply done by specifyng the minimum version of pip in the requirements.txt file) (b) marking this option as 'EXPERIMENTAL' (there is an environment variable for this that I added on another PR) and only invoke if requested and not by default.

I noticed the warning from pip. We should possibly be catching the output from the subprocess call (stderr, stdout) before passing this to the json loader.